wear/src/main/java/io/homeassistant/companion/android/home/views/SelectTemplateTileView.kt

```diff
@@ -0,0 +1,86 @@
+package io.homeassistant.companion.android.home.views
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Devices
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.wear.compose.material.Chip
+import androidx.wear.compose.material.ChipDefaults
+import androidx.wear.compose.material.PositionIndicator
+import androidx.wear.compose.material.Scaffold
+import androidx.wear.compose.material.Text
+import androidx.wear.compose.material.itemsIndexed
+import androidx.wear.compose.material.rememberScalingLazyListState
+import io.homeassistant.companion.android.theme.WearAppTheme
+import io.homeassistant.companion.android.views.ListHeader
+import io.homeassistant.companion.android.views.ThemeLazyColumn
+import io.homeassistant.companion.android.common.R as commonR
+
+@Composable
+fun SelectTemplateTileView(
+    templateTileIds: List<Int?>,
+    onSelectTemplateTile: (tileId: Int?) -> Unit
+) {
+    val scalingLazyListState = rememberScalingLazyListState()
+    WearAppTheme {
+        Scaffold(
+            positionIndicator = {
+                if (scalingLazyListState.isScrollInProgress) {
+                    PositionIndicator(scalingLazyListState = scalingLazyListState)
+                }
+            },
+            timeText = { TimeText(scalingLazyListState = scalingLazyListState) }
+        ) {
+            ThemeLazyColumn(state = scalingLazyListState) {
+                item {
+                    ListHeader(id = commonR.string.template_tiles)
+                }
+                item {
+                    ListHeader(id = commonR.string.template_tile_select)
+                }
+                // TODO: make sure this refreshes whenever a new Template tile is added
+                if (templateTileIds.isEmpty()) {
+                    item {
+                        Text(stringResource(commonR.string.template_tile_no_tiles_yet))
+                    }
+                } else {
+                    itemsIndexed(templateTileIds) { index, templateTileId ->
+                        Chip(
+                            modifier = Modifier
+                                .fillMaxWidth(),
+                            label = {
+                                Text(stringResource(commonR.string.template_tile_n, index + 1))
+                            },
+                            onClick = { onSelectTemplateTile(templateTileId) },
+                            colors = ChipDefaults.secondaryChipColors()
+                        )
+                    }
+                }
```

Comment:
I'm still too new to this to understand how to properly solve this. Whenever a new Template tile is added in the Tile carousel, how do we make sure this view is refreshed with that info? I had to force close the HA Wear app and starting it again for a new Template tile to show up.

It looks like the same problem does not exist for the Shortcut tiles, but I couldn't figure out what I'm doing differently here.

Reply:
So with compose as you are passing them in as a variable `templateTileIds` in the method it should trigger a recomposition anytime the variable changes. I would check to make sure the variable is updating as you would expect. Should work similar to how we load entities on the home page or process state updates.

Reply:
I changed `SelectTemplateTileView` to take `templateTiles: Map<Int?, TemplateTileConfig>` as a parameter instead, and `HomeView` passes in the mutable reference from `MainViewModel`. I thought that would do the trick, but it still did not update automatically 😱